modules/integration/tests-integration/tests-backend/src/test/java/org/wso2/identity/integration/test/oauth2/OAuth2AuthorizationCodeGrantForSubOrgAppsTestCase.java

```diff
@@ -0,0 +1,674 @@
+import static org.wso2.identity.integration.test.utils.OAuth2Constant.AUTHORIZATION_HEADER;
+import static org.wso2.identity.integration.test.utils.OAuth2Constant.BASIC_HEADER;
+import static org.wso2.identity.integration.test.utils.OAuth2Constant.USER_AGENT;
+import static org.wso2.identity.integration.test.utils.OAuth2Constant.ACCESS_TOKEN_ENDPOINT;
+
+/**
+ * Holds the integration tests for sub organization application authorizations when behaving as a tenant.
+ */
+public class OAuth2AuthorizationCodeGrantForSubOrgAppsTestCase extends OAuth2ServiceAbstractIntegrationTest {
+
+    private static final String API_SERVER_BASE_PATH = "api/server/v1";
+    private static final String API_RESOURCE_MANAGEMENT_PATH = "/api-resources";
+    private static final String APPLICATION_MANAGEMENT_PATH = "/applications";
+    private static final String AUTHORIZED_APIS_PATH = "/authorized-apis";
+    public static final String TENANT_PATH = "t/";
+    public static final String PATH_SEPARATOR = "/";
+    private static final String API_RESOURCES = "apiResources";
+    private static final String ID = "id";
+    private static final String POLICY_IDENTIFIER = "policyIdentifier";
+    private static final String RBAC_POLICY = "RBAC";
+    private static final String SCOPES = "scopes";
+    private static final String ORG_END_USER_USERNAME = "alex";
+    private static final String ORG_END_USER_PASSWORD = "Alex@wso2";
+    private static final String ORG_END_USER_EMAIL = "alex@wso2.com";
+    private static final String TEST_ORG_END_USER_USERNAME = "testUser";
+    private static final String TEST_ORG_END_USER_PASSWORD = "TestUser@wso2";
+    private static final String TEST_ORG_END_USER_EMAIL = "testuser@wso2.com";
+    private static final String MGT_APP_AUTHORIZED_API_RESOURCES = "management-app-authorized-apis.json";
+    private static final String USERS = "users";
+    private static final String ORGANIZATION_NAME = "sub001";
+    private static final String ORGANIZATION_HANDLE = "sub001";
+    private static final String ORGANIZATION_TEST_ADMIN_USER = "testadmin@" + ORGANIZATION_HANDLE;
+    private static final String ORGANIZATION_MAIN_APP_NAME = "Sub Organization Application - Auth Code";
+    private static final String ORGANIZATION_TEST_APP_NAME = "Token Test Org App";
+    private static final String SYSTEM_SCOPE = "SYSTEM";
+    private static String CLIENT_ID;
+    private static String CLIENT_SECRET;
+    private static String ORG_APP_CLIENT_ID;
+    private static String ORG_APP_CLIENT_SECRET;
+    private CloseableHttpClient client;
+    private SCIM2RestClient scim2RestClient;
+    private OrgMgtRestClient orgMgtRestClient;
+    private String accessToken;
+    private String orgAppAccessToken;
+    private String switchedM2MToken;
+    private String sessionDataKey;
+    private String orgEndUserId;
+    private String authorizationCode;
+    private String organizationId;
+
```

Comment:
need to execute the test for both super tenant and another tenant mode
Add the relevant data provider and mode executor

Reply:
Done